flink-table/flink-table-common/src/main/java/org/apache/flink/table/data/GenericMapData.java

```diff
@@ -85,12 +85,15 @@ public boolean equals(Object o) {
 			return false;
 		}
 		GenericMapData that = (GenericMapData) o;
-		return Objects.equals(map, that.map);
+		return Arrays.deepEquals(map.keySet().toArray(), that.map.keySet().toArray()) &&
+			Arrays.deepEquals(map.values().toArray(), that.map.values().toArray());
```

Comment:
Not sure about this. `Arrays.deepEquals` will compare elements in order. However, `map.keySet()` doesn't guarentee order. Maybe we can simply use `Objects.equals(map, that.map)` here. Because `byte[]` shouldn't be as map key.

Reply:
Good point, we can do the key check differently. But for values we need to include `byte[]` as well.

Reply:
True. Maybe we need to foreach entries and get values using the entry key.

Reply:
Yes, I will just copy the implementation of `HashMap.equals/HashMap.hashCode` and replace the `equals` call.

Reply:
Updated.